### PR TITLE
Add test cases for ImmediateExecutor.

### PR DESCRIPTION
**Pull request by seedeed**
Motivation:

ImmediateExecutor needs more test cases.

Modification:

Add several test cases for ImmediateExecutor.

Result:

Improve test coverage slightly.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2014 The Netty Project
```
Comment (normanmaurer) [inline common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java]: nit: 2020

File: common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        ImmediateExecutor.INSTANCE.execute(null);
+    }
+
+    @Test
+    public void testExecuteNonNullRunnable() throws Exception {
+        FutureTask<Void> task = new FutureTask<Void>(new Runnable() {
+
+            @Override
+            public void run() {
+
```
Comment (normanmaurer) [inline common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java]: add `// NOOP` or use the following syntax:

`new Runnable() { }`

File: common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            }
+
+        }, null);
+        ImmediateExecutor.INSTANCE.execute(task);
+        assertTrue(task.isDone());
+        assertFalse(task.isCancelled());
+        assertTrue(task.get() == null);
```
Comment (normanmaurer) [inline common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java]: `assertNull(task.get())`

File: common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java (hunk 2 above)
Comment (seedeed) [reply common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java]: got it.

File: common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java (hunk 3 above)
Comment (seedeed) [reply common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java]: got it

File: common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java (hunk 2 above)
Comment (seedeed) [reply common/src/test/java/io/netty/util/concurrent/ImmediateExecutorTest.java]: got it.

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@seedeed thanks a lot!

**Pull request merged**